src/v2/client/handleConfiguredFlow.js

```diff
@@ -0,0 +1,105 @@
+import {
+  getTransactionMeta,
+  getSavedTransactionMeta,
+  saveTransactionMeta,
+  clearTransactionMeta,
+} from './transactionMeta';
+
+async function proceedIfAvailable(idxResponse, remediation) {
+  console.log(idxResponse);
+  const rem = idxResponse.neededToProceed.find(item => item.name === remediation);
+  if (rem) {
+    return await idxResponse.proceed(remediation);
+  }
+  Logger.warn('Expected remediation not found, Org may be misconfigured for this flow');
+  return idxResponse;
+};
+
+async function stepIntoSpecificIdxFlow(idxState, flow='') {
+  console.log('stepIntoSpecificIdxFlow called');
+  try {
+    if ([Enums.LOGIN_FLOW, Enums.DEFAULT_FLOW, Enums.CONTINUE_FLOW].includes(flow)) {
+      // default IDX response from interact is "Login" page/flow. Do nothing
+      return idxState;
+    }
+
+    if (flow === Enums.REGISTRATION_FLOW) {
+      return await proceedIfAvailable(idxState, 'select-enroll-profile');
+    }
+
+    if (flow === Enums.RESET_PASSWORD_FLOW) {
+      return idxState.actions['currentAuthenticator-recover'] ?
+        await idxState.actions['currentAuthenticator-recover']() : idxState;
+    }
+
+    if (flow === Enums.UNLOCK_ACCOUNT_FLOW) {
+      // requires: introspect -> identify-recovery -> select-authenticator-unlock-account
+      return await proceedIfAvailable(idxState,'unlock-account');
+    }
+
+    Logger.warn(`Unknown \`flow\` value: ${flow}`);
+    return idxState;
+  }
+  catch (err) {
+    console.log('stepIntoSpecificIdxFlow Error caught')
+    // catches and handles `Unknown remediation` errors thrown okta-idx-js
+    if (typeof err === 'string' && err.startsWith('Unknown remediation choice')) {
+      Logger.warn(`flow [${flow}] not valid with current Org configurations`);
+      // throw new Errors.InitialFlowError('Unable to proceed to desired flow', flow);
+      return idxState;
+    }
+    else {
+      // do not catch non-`Unknown remediation` errors here
+      throw err;
+    }
+  }
+}
+
+export async function handleConfiguredFlow(originalResp, settings) {
+  const configuredFlow = settings.get('flow');
+
+  if (!configuredFlow || configuredFlow === Enums.CONTINUE_FLOW || configuredFlow === Enums.DEFAULT_FLOW) {
+    return originalResp;
+  }
+
+  let idxState = originalResp;
+
+  const meta = await getTransactionMeta(settings);
+  console.log(meta.flowId, configuredFlow)
+
+  if (meta.flowId && meta.flowId !== configuredFlow) {
+    // configured flow and active flow do not match, abandon active flow, start new (configured) flow
+    Logger.warn(`Canceling current '${meta.flowId}' flow to start '${configuredFlow}' flow`);
+    sessionStorageHelper.removeStateHandle();
+    clearTransactionMeta(settings);
+    idxState = await interact(settings);
+  }
+
+  // attempts to step into the desired flow
+  idxState = await stepIntoSpecificIdxFlow(idxState, configuredFlow);
+
+  // meta could have been mutated since the first `getTransactionMeta` call in this function
```

Comment:
curious what kind of mutations we expect?

Reply:
when the `configuredFlow` !== `meta.flowId` that results in an `interact` call, which returns a new `interaction_handle` and that new handle was being overwritten by the old one before calling `getSavedTransactionMeta` a second time